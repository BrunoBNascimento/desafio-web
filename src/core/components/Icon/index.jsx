import React from 'react';
import PropTypes from 'prop-types';
import './style.scss';

const Icon = ({ icon }) => (
  <i className={`fas fa-${icon}`} />
);

export default Icon;

Icon.propTypes = {
  icon: PropTypes.string.isRequired,
};
