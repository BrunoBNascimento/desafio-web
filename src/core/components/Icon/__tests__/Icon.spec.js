import React from 'react';
import enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Icon from '../';

// tslint:disable-next-line:no-any
enzyme.configure({ adapter: new Adapter() });

describe('Icon', () => {
  it('should render Icon', () => {
    const icon = shallow(<Icon icon="user" />);
    expect(icon).toMatchSnapshot();
  });
});
