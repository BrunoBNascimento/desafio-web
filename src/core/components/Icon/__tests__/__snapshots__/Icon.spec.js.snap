// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon should render Icon 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Icon
    icon="user"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "fas fa-user",
    },
    "ref": null,
    "rendered": null,
    "type": "i",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "fas fa-user",
      },
      "ref": null,
      "rendered": null,
      "type": "i",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
